core/src/main/java/org/infinispan/interceptors/impl/DistCacheWriterInterceptor.java

```diff
@@ -145,6 +147,42 @@ public Object visitReplaceCommand(InvocationContext ctx, ReplaceCommand command)
    }
 
    @Override
+   public Object visitComputeCommand(InvocationContext ctx, ComputeCommand command) throws Throwable {
+      return invokeNextThenApply(ctx, command, (rCtx, rCommand, rv) -> {
+         ComputeCommand computeCommand = (ComputeCommand) rCommand;
+         Object key = computeCommand.getKey();
+         if (!isStoreEnabled(computeCommand) || rCtx.isInTxScope() || !computeCommand.isSuccessful())
+            return rv;
+         if (!isProperWriter(rCtx, computeCommand, computeCommand.getKey()))
+            return rv;
+
+         storeEntry(rCtx, key, computeCommand);
```

Comment:
The state already is in the entry: the command sets `CacheEntry.isRemoved` to false, so you'd just test this one...